python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -936,6 +936,20 @@ def nth(self, n):
         del self.obj._data["__groupbynth_order__"]
         return result
 
+    @_cudf_nvtx_annotate
+    def topk(self, k, bottom=False):
```

Comment:
Their implementation is a bit more involved, since it has tie-breaking rules that necessitate (if `keep != "first"`) introspecting the data rather than just the group size. So I can fill out nlargest/smallest for `keep="first"` and raise otherwise.